### PR TITLE
Quiz_maker

### PR DESCRIPTION
**Pull request by krassovki**
- added main
- added tests (all of them passed)
- test_get_given_amount_of_random_questions_with_wrong_keys_of_questions
  (line 54) trying to run toooo much time

### CODE REVIEW

File: Misha/Files/quiz_maker/quiz_maker_main.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                        rand_index = random.randint(-1, number_of_questions_to_remove)
+                    questions.pop(rand_index)
+                except KeyError:
+                    raise KeyError
+            return questions
+
+
+def get_answers_list(path):
+    if not os.path.exists(path):
+        raise FileNotFoundError('File not found')
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_main.py]: Do you need this error?

File: Misha/Files/quiz_maker/quiz_maker_main.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            random.seed(time.time())
+            number_of_questions_to_remove = len(questions) - number_of_questions
+            for rand_value in range(number_of_questions_to_remove):
+                rand_index = random.randint(0, number_of_questions_to_remove)
+                try:
+                    while questions.get(rand_index) is None:
+                        rand_index = random.randint(-1, number_of_questions_to_remove)
+                    questions.pop(rand_index)
+                except KeyError:
+                    raise KeyError
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_main.py]: Do you need this raise?

File: Misha/Files/quiz_maker/quiz_maker_main.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        key_answers = ''
+        for line in f:
+            key_answers += line
+        f.close()
+        key_answers = key_answers.split('\n')
+        return key_answers
+
+
+def get_needed_answers(key_answers, questions_keys):
+    if type(key_answers) != list or type(questions_keys) != list:
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_main.py]: No need to check types here

File: Misha/Files/quiz_maker/quiz_maker_test.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        with self.assertRaises(TypeError) as context:
+            self.assertRaises(TypeError, get_needed_answers(key_answers, questions_keys))
+        exception_message = str(context.exception)
+        self.assertEqual('Given not list', exception_message)
+
+    def test_get_needed_answers_with_wrong_key_answers(self):
+        key_answers = 1488
+        questions_keys = [0, 1, 2, 3]
+        with self.assertRaises(TypeError) as context:
+            self.assertRaises(TypeError, get_needed_answers(key_answers, questions_keys))
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_test.py]: You have two same asserts here

File: Misha/Files/quiz_maker/quiz_maker_test.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                        'Бьярне Страусструп\nг. Илон Маск',
+                     3: 'Какое пиво самое крепкое?\nа. Жиуглевское\nб. Тетерев\nв. Московское\nг. Опилля'}
+        questions = get_given_amount_of_random_questions(number_of_questions, questions)
+        self.assertEqual(number_of_questions, len(questions))
+
+    def test_get_needed_answers_with_wrong_questions_keys(self):
+        key_answers = ['в', 'а', 'а', 'г']
+        questions_keys = 'string'
+        with self.assertRaises(TypeError) as context:
+            self.assertRaises(TypeError, get_needed_answers(key_answers, questions_keys))
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_test.py]: You have two same asserts here

File: Misha/Files/quiz_maker/quiz_maker_test.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    get_needed_answers, \
+    count_points
+
+
+class ZipTest(unittest.TestCase):
+
+    def test_get_questions_with_wrong_paths(self):
+        path = 'wrong_path.txt'
+        with self.assertRaises(FileNotFoundError) as context:
+            self.assertRaises(FileNotFoundError, get_questions(path))
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_test.py]: You have two same asserts here

File: Misha/Files/quiz_maker/quiz_maker_test.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        self.assertEqual('File not found', exception_message)
+
+    def test_get_questions_with_right_paths(self):
+        path = 'questions.txt'
+        expected_questions = {0: 'Сколько лет длилась столетняя война?\na. 100\nб. 998\nв. 113\nг. 228',
+                              1: 'Как звали Гитлера?\nа. Адольф\nб. Геннадий\nв. Васыль\nг. Григорий',
+                              2: 'Кто пидумал современную архитектуру ЭВМ?\nа. Джон фон-Нейман\nб. Карл Цузе\nв. '
+                                 'Бьярне Страусструп\nг. Илон Маск',
+                              3: 'Какое пиво самое крепкое?\nа. Жиуглевское\nб. Тетерев\nв. Московское\nг. Опилля'}
+        questions = get_questions(path)
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_test.py:25]: add tests for this method please

File: Misha/Files/quiz_maker/quiz_maker_test.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        path = 'wrong_path.txt'
+        with self.assertRaises(FileNotFoundError) as context:
+            self.assertRaises(FileNotFoundError, get_answers_list(path))
+        exception_message = str(context.exception)
+        self.assertEqual('File not found', exception_message)
+
+    def test_get_answer_list_with_right_paths(self):
+        path = 'answers.txt'
+        expected_answers = ['в', 'а', 'а', 'г']
+        answers = get_answers_list(path)
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_test.py:31]: Add tests for this method please

File: Misha/Files/quiz_maker/quiz_maker_test.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        self.assertEqual('Given not dict', exception_message)
+
+    def test_get_given_amount_of_random_questions_with_right_input(self):
+        number_of_questions = 3
+        questions = {0: 'Сколько лет длилась столетняя война?\na. 100\nб. 998\nв. 113\nг. 228',
+                     1: 'Как звали Гитлера?\nа. Адольф\nб. Геннадий\nв. Васыль\nг. Григорий',
+                     2: 'Кто пидумал современную архитектуру ЭВМ?\nа. Джон фон-Нейман\nб. Карл Цузе\nв. '
+                        'Бьярне Страусструп\nг. Илон Маск',
+                     3: 'Какое пиво самое крепкое?\nа. Жиуглевское\nб. Тетерев\nв. Московское\nг. Опилля'}
+        questions = get_given_amount_of_random_questions(number_of_questions, questions)
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_test.py:65]: Add tests for this function

File: Misha/Files/quiz_maker/quiz_maker_main.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+def get_questions(path):
+    if not os.path.exists(path):
+        raise FileNotFoundError('File not found')
+    else:
+        f = open(path, 'r', encoding='utf-8')
+        text = ''
+        for line in f:
+            text += line
+        f.close()
```
Comment (glock3) [inline Misha/Files/quiz_maker/quiz_maker_main.py]: Please close file in final block or use with-resource statement

**Review by glock3 — CHANGES_REQUESTED** [review]

File: Misha/Files/quiz_maker/quiz_maker_main.py (hunk 1 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_main.py]: Yes, to check if file exists

File: Misha/Files/quiz_maker/quiz_maker_main.py (hunk 2 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_main.py]: Yes, to check if we have an error with dict keys

File: Misha/Files/quiz_maker/quiz_maker_main.py (hunk 3 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_main.py]: okay(

File: Misha/Files/quiz_maker/quiz_maker_test.py (hunk 4 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_test.py]: corrected

File: Misha/Files/quiz_maker/quiz_maker_test.py (hunk 5 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_test.py]: corrected

File: Misha/Files/quiz_maker/quiz_maker_test.py (hunk 6 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_test.py]: corrected

File: Misha/Files/quiz_maker/quiz_maker_test.py (hunk 7 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_test.py:25]: done in this version

File: Misha/Files/quiz_maker/quiz_maker_test.py (hunk 8 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_test.py:31]: done in this version

File: Misha/Files/quiz_maker/quiz_maker_test.py (hunk 9 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_test.py:65]: done in this version

File: Misha/Files/quiz_maker/quiz_maker_main.py (hunk 10 above)
Comment (krassovki) [reply Misha/Files/quiz_maker/quiz_maker_main.py]: why cant i close it after finishin using it?

**Review by glock3 — APPROVED** [review]

**Pull request merged**